src/nbl/video/CVulkanPhysicalDevice.h

```diff
@@ -101,72 +100,163 @@ class CVulkanPhysicalDevice final : public IPhysicalDevice
+        // Todo(achal): not sure yet, how would I handle multiple platforms without making
+        // this function templated
+        VkSurfaceKHR vk_surface = static_cast<const CSurfaceVulkanWin32*>(surface)->getInternalObject();
 
-        std::vector<ISurface::E_PRESENT_MODE> result(count);
-        for (uint32_t i = 0u; i < count; ++i)
+        VkSurfaceCapabilitiesKHR vk_surfaceCapabilities;
+        if (vkGetPhysicalDeviceSurfaceCapabilitiesKHR(m_vkPhysicalDevice, vk_surface,
+            &vk_surfaceCapabilities) != VK_SUCCESS)
         {
-            result[i] = ISurfaceVK::getPresentMode(vk_presentModes[i]);
+            return false;
         }
 
-        return result;
+        capabilities.minImageCount = vk_surfaceCapabilities.minImageCount;
+        capabilities.maxImageCount = vk_surfaceCapabilities.maxImageCount;
+        capabilities.currentExtent = vk_surfaceCapabilities.currentExtent;
+        capabilities.minImageExtent = vk_surfaceCapabilities.minImageExtent;
+        capabilities.maxImageExtent = vk_surfaceCapabilities.maxImageExtent;
+        capabilities.maxImageArrayLayers = vk_surfaceCapabilities.maxImageArrayLayers;
+        // Todo(achal)
+        // VkSurfaceTransformFlagsKHR       supportedTransforms;
+        // VkSurfaceTransformFlagBitsKHR    currentTransform;
+        // VkCompositeAlphaFlagsKHR         supportedCompositeAlpha;
+        capabilities.supportedUsageFlags = static_cast<asset::IImage::E_USAGE_FLAGS>(vk_surfaceCapabilities.supportedUsageFlags);
+
+        return true;
     }
-    
-    uint32_t getMinImageCountForSurface(const ISurface* surface) const override
+
+    bool isSwapchainSupported() const override
     {
-        // Todo(achal): This is problematic, if passed `surface` isn't a vulkan surface
-        const ISurfaceVK* vk_surface = static_cast<const ISurfaceVK*>(surface);
+        constexpr uint32_t MAX_DEVICE_EXTENSIONS_COUNT = 250u;
+
+        uint32_t availableExtensionCount;
+        VkResult retval = vkEnumerateDeviceExtensionProperties(m_vkPhysicalDevice, NULL, &availableExtensionCount, NULL);
 
-        VkSurfaceCapabilitiesKHR surfaceCapabilities;
-        vkGetPhysicalDeviceSurfaceCapabilitiesKHR(m_vkphysdev, vk_surface->m_surface,
-            &surfaceCapabilities);
+        // Todo(achal): Would there be a need to handle VK_INCOMPLETE separately?
+        if ((retval != VK_SUCCESS) && (retval != VK_INCOMPLETE))
+            return false;
+
+        assert(availableExtensionCount <= MAX_DEVICE_EXTENSIONS_COUNT);
+
+        VkExtensionProperties availableExtensions[MAX_DEVICE_EXTENSIONS_COUNT];
+        retval = vkEnumerateDeviceExtensionProperties(m_vkPhysicalDevice, NULL, &availableExtensionCount, availableExtensions);
+
+        // Todo(achal): Would there be a need to handle VK_INCOMPLETE separately?
+        if ((retval != VK_SUCCESS) && (retval != VK_INCOMPLETE))
+            return false;
```

Comment:
you should cache this

Reply:
preferraby in a `core::map<std::string>` or better `core::array<bool> extensions[EVE_COUNT]` where each vulkan device extension has a corresponding enum